### PR TITLE
Added new module for basic security authentication

### PR DESCRIPTION
**Pull request by nbpalomino**

### CODE REVIEW

File: basic-security-client/BasicSecurityStrategy.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+function getUserFromBasicAuth(req) {
+  var authHeader = req.headers['authorization'];
+  console.log("Authorization Header: "+ authHeader);
+
+  if(authHeader) {
+    var [type, token] = authHeader.split(' ');
+    console.log("Authorization Type: "+ type);
+    console.log("Authorization Token: "+ token);
+    //var [user, password] = atob(token).split(':');
```
Comment (jrichardsz) [inline basic-security-client/BasicSecurityStrategy.js]: delete comments!

File: application.json.template [hunk 2]
```diff
@@ -1,7 +1,14 @@
 {
   "server": {
     "security": {
-      "enable": false
+      "enable": false,
+      "npmModule": {
+        "name": "basic-security-client"
+      },
+      "configModule": {
+        "user": "admin",
```
Comment (jrichardsz) [inline application.json.template]: environment vars are easy to use:

"user": "${AUTH_USER}"
"password": ${AUTH_PASSWORD}

This must be in the documentation.

File: basic-security-client/BasicSecurityStrategy.js (hunk 1 above)
Comment (nbpalomino) [reply basic-security-client/BasicSecurityStrategy.js]: Fixed...

File: application.json.template (hunk 2 above)
Comment (nbpalomino) [reply application.json.template]: Fixed...

**Pull request merged**